internal/internal_test.go

```diff
@@ -1272,12 +1274,129 @@ func TestGetParsedBody(t *testing.T) {
 		}
 	  }`
 
-	_, _, err := getParsedBody(createCheckRequest(requestContentTypeJSONInvalid))
+	path := []interface{}{}
+	protoDescriptorPath := ""
+
+	_, _, err := getParsedBody(createCheckRequest(requestContentTypeJSONInvalid), path, protoDescriptorPath)
 	if err == nil {
 		t.Fatal("Expected error but got nil")
 	}
 }
 
+func TestGetParsedBodygRPC(t *testing.T) {
+
+	requestValidExample := `{
+		"attributes": {
+		  "request": {
+			"http": {
+			  "headers": {
+				"content-type": "application/grpc"
+			  },
+			  "method": "POST",
+			  "path": "/Example.Test.GRPC.ProtoServiceIExampleApplication/RegisterExample",
+			  "protocol": "HTTP/2",
+			  "raw_body": "AAAAADYKFgoHCgVFUlJPUhILCglTZWNOdW1iZXISHAoMCgpCb2R5IHZhbHVlEgwKCk5hbWUgVmFsdWU=",
+			}
+		  },
+		  "parsed_path": [
+			"Example.Test.GRPC.ProtoServiceIExampleApplication",
+			"RegisterExample"
+		  ]
+		}
+	  }`
+
+	requestInvalidRawBodyExample := `{
+		"attributes": {
+		  "request": {
+			"http": {
+			  "headers": {
+				"content-type": "application/grpc"
+			  },
+			  "method": "POST",
+			  "path": "/Example.Test.GRPC.ProtoServiceIExampleApplication/RegisterExample",
+			  "protocol": "HTTP/2"
+			}
+		  },
+		  "parsed_path": [
+			"Example.Test.GRPC.ProtoServiceIExampleApplication",
+			"RegisterExample"
+		  ]
+		}
+	  }`
+
+	requestInvalidParsedPathExample := `{
+		"attributes": {
+		  "request": {
+			"http": {
+			  "headers": {
+				"content-type": "application/grpc"
+			  },
+			  "method": "POST",
+			  "protocol": "HTTP/2",
+			  "raw_body": "AAAAAC0KDQoHCgVFUlJPUhICCAESHAoMCgpCb2R5IHZhbHVlEgwKCk5hbWUgVmFsdWU="
+			}
+		  }
+		}
+	  }`
+
+	expectedObject1 := map[string]interface{}{}
+	expectedObject1["Body"] = "Body value"
+	expectedObject1["Name"] = "Name Value"
+
+	expectedObject2 := map[string]interface{}{}
+	expectedObject2["SeverityNumber"] = "SecNumber"
+	expectedObject2["SeverityText"] = "ERROR"
+
+	expectedObject := map[string]interface{}{}
+	expectedObject["Data"] = expectedObject1
+	expectedObject["Metadata"] = expectedObject2
+
+	//grpcserver := testAuthzServer(&testPlugin{}, false)
+
+	//grpcserverWithConfig := testAuthzServer(&testPlugin{}, false)
+	//grpcserverWithConfig.cfg.ProtoDescriptor = `../test/files/example.pb`
+
+	protoDescriptorPath := `../test/files/example.pb`
+
+	tests := map[string]struct {
+		input               *ext_authz.CheckRequest
+		want                interface{}
+		isBodyTruncated     bool
```

Comment:
I don't think we currently support proper detection of a truncated protobuf body, do we? This could be a follow-up, I suppose.

Reply:
currently i dont think so